[MM-55998] incorporate more findings from community (#6822)
mattermost/docs: source/deploy/postgres-migration.rst

.. note::
   If you are using MySQL v8: Due to a `known bug <https://github.com/dimitri/pgloader/issues/1183>`__ in pgLoader compiled binaries, you need to compile pgLoader from the source. Please follow the steps `here <https://pgloader.readthedocs.io/en/latest/install.html#build-from-sources>`__ to build from the source.

> .. note::

Reviewer: Please combine these 2 notes together formatted as bullets.

>    We have received reports about that the pgloader docker image can be limited in terms of memory resources. Please use pgloader directly instead of a docker container. 

Reviewer: ```suggestion
   We have received reports that the pgloader Docker image can be limited in terms of memory resources. Please use pgloader directly instead of a Docker container. 
```

-  Determine the migration window needed. This process requires you to stop the Mattermost Server during the migration.
-  See the `schema-diffs <#schema-diffs>`__ section to ensure data compatibility between schemas.
-  Prepare your PostgreSQL environment by creating a database and user. See the `database </install/prepare-mattermost-database.html>`__ documentation for details.

> -  If you are planning to run an iterative migration (running the pgloader several times), please see the :ref:`designated section<Iterative migrations>`.

Reviewer: Has this link been verified as working?

Author: I literally pushed the PR for this :D There is a problem locally as Python causing an OOM :

![Screenshot 2023-12-06 at 15 34 59](https://github.com/mattermost/docs/assets/2153367/a6907509-3621-4310-8c43-1107ef9230b0)

Reviewer: The Python issue is known. Docs doesn't have a fix for it yet. The workaround I use is to kill those processes.

   dbcmp --source "${MYSQL_DSN}" --target "${POSTGRES_DSN}" --exclude="db_migrations,systems"

Iterative migrations
-------------------

> Thera are several steps in the pgloader configuration file that assumes that migration will take place at one go. If you are planning to run the migration over and over again, please do the changes defined below:

Reviewer: ```suggestion
Thera are several steps in the pgloader configuration file that assume migration will take place in one go. If you are planning to run the migration over and over again, please complete the changes defined below:
```
